### ISSUE TITLE
Migrate to FSDP2?

### ISSUE THREAD

**Issue by wconstab** [volcengine/verl#1156]
The PyTorch team is trying to migrate users from FSDP to FSDP2 (a.k.a. fully_shard).  verl is a popular and growing source of new usage of FSDP1 and we want to ask if the verl team is open to migrating.  We may be able to help with migration work and it would also be useful to know if there are specific features of FSDP1 you are relying on or if FSDP2 is already good enough.   Thanks!

**Comment by eric-haibin-lin** [comment]
yes we're quite open to migration. verl currently just use the common stuff, mixed precision, wrap policy, gradient accumulation.

**Comment by wconstab** [comment]
Thanks to @lxg2015 for initiating this work! https://github.com/volcengine/verl/pull/1026

**Comment by wconstab** [comment]
Happy to see that https://github.com/volcengine/verl/pull/1026 is merged!  I wonder if folks have an idea about how/when to switch the default for VERL to use fsdp2? cc  @PeterSH6 @eric-haibin-lin

**Comment by eric-haibin-lin** [comment]
> Happy to see that https://github.com/volcengine/verl/pull/1026 is merged!  I wonder if folks have an idea about how/when to switch the default for VERL to use fsdp2? cc  @PeterSH6 @eric-haibin-lin 

We can prioritize the fsdp2 implementation when doing #1371

**Comment by Kirrito-k423** [comment]
### Scheduled issue cleanup / 定时 issue 清理

中文：该 issue 命中定时清理规则：最近一次非当前运行账号的公开回复或引用已经超过 6 个月。因此按定时清理标准先关闭。若作者或社区成员仍能提供新的复现、证据或需求背景，欢迎继续评论请求重开，或创建带有最新信息的新 issue。

证据：
- Issue: https://github.com/verl-project/verl/issues/1156
- 最近一次非当前运行账号的公开回复或公开引用是 `2025-05-03T03:33:16Z` UTC（mentioned by @eric-haibin-lin），截至 2026-07-05 已超过 6 个月。
- 已检查 comments 来源：https://api.github.com/repos/verl-project/verl/issues/1156/comments
- 已检查 timeline 来源：https://api.github.com/repos/verl-project/verl/issues/1156/timeline

English: This issue matches the scheduled cleanup rule: the latest public reply or reference by an account other than the current runner is older than 6 months. Therefore, this is being closed under the scheduled cleanup policy. If the author or community members can still provide a fresh reproduction, evidence, or requirement context, please comment to request reopening or open a new issue with updated information.

Evidence:
- Issue: https://github.com/verl-project/verl/issues/1156
- The latest public reply or public reference from an account other than the current runner was `2025-05-03T03:33:16Z` UTC (mentioned by @eric-haibin-lin), which is more than 6 months before 2026-07-05.
- Checked comments source: https://api.github.com/repos/verl-project/verl/issues/1156/comments
- Checked timeline source: https://api.github.com/repos/verl-project/verl/issues/1156/timeline

This response was generated by a scheduled Codex GPT5.5 issue processing task.
